src/Symfony/Component/Security/Core/User/LdapUserProvider.php

```diff
@@ -113,30 +118,31 @@ public function supportsClass($class)
      */
     protected function loadUser($username, Entry $entry)
     {
-        $password = $this->getPassword($entry);
+        $password = $this->getAttributeValue($entry, $this->passwordAttribute);
 
         return new User($username, $password, $this->defaultRoles);
     }
 
     /**
-     * Fetches the password from an LDAP entry.
+     * Fetches a required unique attribute value from an LDAP entry.
      *
      * @param null|Entry $entry
+     * @param string     $attribute
      */
-    private function getPassword(Entry $entry)
+    private function getAttributeValue(Entry $entry, $attribute)
     {
-        if (null === $this->passwordAttribute) {
+        if (null === $attribute) {
```

Comment:
@csarrazi Ok, I have misunderstood what you mean, my bad. However, I think there is a little BC break risk, if users are bypassing the `$uidKey` parameter to write a custom filter (in that case, they would set the `$uidKey` to null and the new implementation would throw an exception, the old one does not).

I think it would be weird to use the user provider without a `$uidKey` but there seems to be a risk as the current implementation allows it so I want your confirmation before performing this modification.